sdk/core/core-http/lib/util/xml.ts

```diff
@@ -35,3 +38,152 @@ export function parseXML(str: string): Promise<any> {
     }
   });
 }
+
+export async function deserializeAtomXmlToJson(body: string): Promise<any> {
+  let parsed;
+  const parser = new xml2js.Parser(_getDefaultSettingsForAtomXmlOperations());
+  await parser.parseString(_removeBOM(body.toString()), function(err: any, parsedBody: any) {
```

Comment:
await isn't adding anything here since `parseString` doesn't return a promise. You'd need to convert this to
```typescript
await new Promise((resolve, reject) => {
  parser.parseString(_removeBOM(body.toString()), function(err: Error, parsedBody: any) {
    if (err) {
      reject(err);
    } else {
      resolve(parsedBody);
    }
  });
});
```

Reply:
Got it, I thought we suspected that the implementation would get promisified later.

Took a closer look at the `parseString` implementation with @bterlson and it looks like the implementation is synchronous but only if the `options.async` flag is not set. 
Agree with safeguarding against this getting set, and thanks for the interesting catch !